features/query/timeout_due_to_no_active_workers/feature.go

```diff
@@ -0,0 +1,47 @@
+package timeout_due_to_no_active_workers
+
+import (
+	"context"
+	"go.temporal.io/api/serviceerror"
+	"go.temporal.io/sdk-features/harness/go/harness"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/workflow"
+	"time"
+)
+
+var Feature = harness.Feature{
+	Workflows: Workflow,
+	CheckResult: func(ctx context.Context, r *harness.Runner, run client.WorkflowRun) error {
+		// Shut off the worker
+		r.Worker.Stop()
+
+		oneSecCtx, cf := context.WithDeadline(ctx, time.Now().Add(time.Second))
+		defer cf()
+		_, err := r.Client.QueryWorkflow(oneSecCtx, run.GetID(), run.GetRunID(), "somequery")
+		r.Require.Error(err)
+		_, ok := err.(*serviceerror.DeadlineExceeded)
+		r.Require.True(ok)
+
+		err = r.RestartWorker()
+		if err != nil {
+			return err
+		}
+
+		// Finish the wf
+		err = r.Client.SignalWorkflow(ctx, run.GetID(), run.GetRunID(), "finish", nil)
```

Comment:
Just let the workflow finish immediately and run `CheckResultDefault` before you even start querying. Might as well run the queries on a closed workflow, you don't have to restart the worker.

Reply:
True, but, this works and exists now - the restarting functionality is likely to be useful for something else later.